File: src/satellite-autonomy-service/Sentinel.Core/Entities/Command.cs
using Sentinel.Core.Enums;

namespace Sentinel.Core.Entities;

public sealed class Command
{
    public Guid Id { get; set; }
    public Guid SatelliteId { get; set; }
    public Guid? MissionId { get; set; }
    public string Type { get; set; } = string.Empty;
    public string? PayloadJson { get; set; }
    public int Priority { get; set; }
    public int TtlSec { get; set; }
    public CommandStatus Status { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? ClaimedAt { get; set; }
    public DateTime? ExecutedAt { get; set; }
}

File: src/ground-ui-service/src/types/command.ts
export interface CommandDto {
  id: string;
  satelliteId: string;
  missionId?: string | null;
  type: string;
  payloadJson?: string | null;
  priority: number;
  ttlSec: number;
  status: string;
  createdAt: string;
  claimedAt?: string | null;
  executedAt?: string | null;
}

export interface CommandCreateRequest {
  type: string;
  priority: number;
  ttlSec: number;
  payloadJson?: string | null;
}
